### ISSUE TITLE
mth5 tests fail with version 0.4.8 of mth5 package

### ISSUE THREAD

**Issue by jameswilburlewis** [spedas/pyspedas#1044]
Our github tests have recently started failing in the mth5 test suite.  On my laptop, the tests were passing with mth5 version 0.4.6.  The github tests are using mth5 version 0.4.8.   When I upgraded to that version in my environment, I started seeing the same failures:

```
Error
Traceback (most recent call last):
  File "/Users/jwl/PycharmProjects/pyspedas/pyspedas/mth5/tests/test_load_fdsn.py", line 118, in test02_load_fdsn_basic
    load_fdsn(network="4P", station="REU49", trange=[date_start, date_end])
  File "/Users/jwl/PycharmProjects/pyspedas/pyspedas/mth5/load_fdsn.py", line 264, in load_fdsn
    run_ts = run_ts.calibrate()
  File "/Users/jwl/PycharmProjects/pyspedas/venv/lib/python3.9/site-packages/mth5/timeseries/run_ts.py", line 1051, in calibrate
    new_run.add_channel(calibrated_ch_ts)
  File "/Users/jwl/PycharmProjects/pyspedas/venv/lib/python3.9/site-packages/mth5/timeseries/run_ts.py", line 765, in add_channel
    raise ValueError(msg)
ValueError: Channel sample rate is not correct, current 0.0 input 1.0


```

I think mth5 was the only package that was reinstalled in the upgrade.

Reverting to mth5 version 0.4.6, the tests pass again. Upgrading to 0.4.7, the tests still pass. So it seems that the problem only occurs with 0.4.8.

**Comment by xandrd** [comment]
Hi @jameswilburlewis,

I suggest we lock the version of `mth5` to 0.4.6 (or possibly 0.4.7, I’ll need to confirm). The `mth5` package was being developed with our specific requirements in mind, but the latest updates don’t necessarily account for our needs. I’ve encountered this issue before, and we were able to resolve it at the time. However, since the grant has ended, it might be more challenging to request direct support for our modifications moving forward.

**Comment by jameswilburlewis** [comment]
mth5 isn't in our requirements.txt, so we'd need to add it if we want to constrain the versions that are compatible with pyspedas.   I don't think there are any problems installing those mth5 releases on the platforms that pyspedas supports, so maybe that's not so bad.   We can open an issue with them about the latest version and see if they're willing to address it, or maybe they'll be able to suggest a workaround.

**Comment by jameswilburlewis** [comment]
I've opened an issue at the mth5 repo: https://github.com/kujaku11/mth5/issues/255

It might be a one-line fix....I've forked their repo and will submit a PR if I can get it working again.

**Comment by jameswilburlewis** [comment]
It looks like they committed a fix yesterday which fixes the problem....hopefully they'll release it soon!  I contributed a PR to resolve some SyntaxWarnings for invalid escape sequences in Python 3.12 and above.

**Comment by jameswilburlewis** [comment]
I made some slight changes to the logic to detect when mth5 is installed, and to recommend version 0.4.7 until a new release fixing the problem in 0.4.8 is available.

**Comment by jameswilburlewis** [comment]
Confirmed fixed in mth5 version 0.4.9
